### PR TITLE
Emily openapi client test suite - (OpenAPI Client 3/3)

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Description

Closes: #394
Related: #50 

## Changes

Add a test suite for the Emily API that uses the OpenAPI client instead of a bespoke client like in the original integration tests. This is a better model for testing the API than we previously had because it directly mimics how outside users would use the Emily API client.

## Testing Information

To run this you need to run a number of commands concurrently. #505 will make these unnecessary, this is just to test this current code:

- Terminal 1: `make emily-integration-env-up`
- Terminal 2: `make emily-server-watch`
- Terminal 3: `make emily-integration-test`

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: emily/client-tests/tests/integration/chainstate.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            .expect("Received an error after making a valid set chainstate api call.");
+
+    let gotten_post_reorg_chaintip = apis::chainstate_api::get_chain_tip(&configuration)
+        .await
+        .expect("Received an error after making a valid get chaintip api call.");
+
+    // Assert.
+    // --------
+    assert_eq!(expected_post_reorg_chaintip, created_reorged_chainstate);
+    assert_eq!(expected_post_reorg_chaintip, gotten_post_reorg_chaintip);
```
Comment (djordon) [inline emily/client-tests/tests/integration/chainstate.rs:125]: At the end of this test we check that the chain tip is at the `reorg_height`, but isn't that incorrect? Why doesn't the chain tip always point to the chain with the max length? I think this implies that a new block with a height that is lower than the current max height is not really a reorg yet.

File: emily/client-tests/tests/integration/withdrawal.rs [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        .collect::<Vec<_>>();
+
+    expected_withdrawal_infos.sort_by(arbitrary_withdrawal_info_partial_cmp);
+    gotten_withdrawal_infos.sort_by(arbitrary_withdrawal_info_partial_cmp);
+    assert_eq!(expected_withdrawal_infos, gotten_withdrawal_infos);
+}
+
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn update_deposits() {
```
Comment (djordon) [inline emily/client-tests/tests/integration/withdrawal.rs:196]: nit: typo, should be `update_withdrawals` right?

File: emily/client-tests/tests/integration/deposit.rs [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+                "Received an error after making a valid get deposits for transaction api call.",
+            );
+
+    // Assert.
+    // -------
+    // Expect the deposits to be sorted by output index.
+    // TODO(506): Reverse this order of deposits for this specific api call.
+    expected_deposits.sort_by(|a, b| {
+        b.bitcoin_tx_output_index
+            .partial_cmp(&a.bitcoin_tx_output_index)
```
Comment (djordon) [inline emily/client-tests/tests/integration/deposit.rs:215]: This can be just `.cmp`

**Review by djordon — COMMENTED** [review]
This largely looks good. I had a question about the behavior after the API is notified about a block with lower height that is now part of the longest chain following a reorg. The behavior seems unintuitive to me. These tests just check that the behavior so not really a blocker per se.

Also, these tests can be included in the integration tests in the emily handler, so I think that they should be. Cargo thinks `emily-client` is an independent crate, even before we do `make build` so including these tests in the emily handler is no worse than having these tests in a separate crate from cargo's perspective.

File: emily/client-tests/tests/integration/chainstate.rs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        stacks_block_height: height,
+    }
+}
+
+/// An arbitrary fully ordered partial cmp comparator for Chainstate.
+/// This is useful for sorting vectors of chainstates so that vectors with
+/// the same elements will be considered equal in a test assert.
+fn arbitrary_chainstate_partial_cmp(a: &Chainstate, b: &Chainstate) -> Ordering {
+    let a_str: String = format!("{}-{}", a.stacks_block_hash, a.stacks_block_height);
+    let b_str: String = format!("{}-{}", b.stacks_block_hash, b.stacks_block_height);
```
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply emily/client-tests/tests/integration/chainstate.rs:23]: Neato!

File: emily/client-tests/tests/integration/chainstate.rs (hunk 1 above)
Comment (AshtonStephens) [reply emily/client-tests/tests/integration/chainstate.rs:125]: It actually is a reorg! The API follows the event observer, and the event observer will only ever report a burn block if it's now considered the canonical chain tip. The moment there's a conflicting block that's lower than the existing one it should be considered a reorg.

**Comment by AshtonStephens** [comment]
Moving this into

**Pull request closed without merging**